vignettes/fitting.Rmd

```diff
@@ -0,0 +1,362 @@
+
+```{r}
+filter <- dust2::dust_filter_create(sir(), 0, data, n_particles = 1000)
+likelihood <- dust2::dust_filter_mcstate(filter, sir_packer)
+vcv <- matrix(c(0.0005, 0.0003, 0.0003, 0.0003), 2, 2)
+sampler <- mcstate2::mcstate_sampler_random_walk(vcv)
+samples <- mcstate2::mcstate_sample(posterior, sampler, 1000, 
+                                    initial = sir_packer$pack(pars))
+```
+
+The likelihood now quickly rises up to a stable range and is clearly mixing:
+
+```{r}
+plot(samples$density, type = "l")
+```
+
+The parameters `beta` (per-contact transmission rate) and `gamma` (recovery rate) are strongly correlated
+
+```{r}
+plot(t(drop(samples$pars)), pch = 19, col = "#00000055")
+```
+
+# Deterministic models from stochastic
+
+Another way of fitting this model is to simply throw away the stochasticity.  In the model above we have the lines
+
+```r
+  n_SI <- Binomial(S, p_SI)
+  n_IR <- Binomial(I, p_IR)
+```
+
+which are the stochastic portion of the model.  Each time step we compute the number of individuals who make the transition from `S` to `I` and from `I` to `R` by sampling from the binomial distribution.  We can replace these calls by their expectations (so effectively making `n_SI = S * p_SI`) by running the model in "deterministic" mode.
+
+This simplification of the stochastic model can be seen as taking expectations of the underlying random process, but there's no reason to expect that this represents the mean of the whole model ($E[f(x)] \neq f(E[x])$, at least generally).
+
+We have found these simplifications useful:
+
+* They are not stochastic, so you can use adaptive MCMC or other more efficient algorithms
+* They are orders of magnitude faster, because instead of running 100s or thousands of particles per likelihood evaluation you just run one
+* The region of high probability density of the deterministic model is often within the (broader) region of high probability density of the stochastic model, so you can use these models to create reasonable starting parameter values for your chains
+* The signs and relative magnitudes of the covariances among parameters are often similar between the deterministic and stochastic model, so you can use the deterministic model to estimate a variance-covariance matrix for your stochastic model -- though you will need to increase all quantities in it
+
+Obviously, this approximation comes with costs though:
+
+* You no longer have integer valued quantities from the expectations of samples in your discrete distributions, so you have to think about fractional individuals
+* The model can no longer account for stochastic effects, e.g., at low population sizes.  This can make the model overly rigid, and it may poorly account for observed patterns
+* The fixed `dt` approach is a [first order Euler solver](https://en.wikipedia.org/wiki/Euler_method) which offers few stability guarantees, and this will differ from an system of ODEs solved with a better ODE solver
```

Comment:
```suggestion
* The fixed `dt` approach is a [first order Euler solver](https://en.wikipedia.org/wiki/Euler_method) which offers few stability guarantees, and this will differ from a system of ODEs solved with a better ODE solver
```